atlas/templates/Jenkinsfile.gotmpl

```diff
@@ -11,6 +15,8 @@ pipeline {
   environment {
     GOPATH = "$WORKSPACE"
     DIRECTORY = "src/github.com/{{ if .Root }}{{ .Root }}/{{ end }}{{ .Name }}"
+    GIT_COMMIT = sh('git describe --dirty=-unsupported --always --tags --long || echo pre-commit').trim()
+    IMAGE_VERSION = "${GIT_COMMIT}-j${BUILD_NUMBER}"
```

Comment:
To me, images are immutable. This should just be the git commit. I expect the shared library to handle this problem for us. We wouldn't be pushing different binaries to the same image location, that would be a process failure.

Reply:
We were just trying to leverage the existing conventions - the build number is informative, but you're right it's not required for uniqueness.